pkg/volume/portworx/portworx_util.go

```diff
@@ -0,0 +1,276 @@
+	hostName := m.plugin.host.GetHostName()
+	client, err := util.osdClient(hostName)
+	if err != nil {
+		return "", err
+	}
+
+	devicePath, err := client.Attach(m.volumeID)
+	if err != nil {
+		if err == osdvolume.ErrVolAttachedOnRemoteNode {
+			// Volume is already attached to node.
+			glog.V(2).Infof("Attach operation is unsuccessful. Volume %q is already attached to another node.", m.volumeID)
+			return "", err
+		}
+		glog.V(2).Infof("AttachVolume on %v failed with error %v", m.volumeID, err)
+		return "", err
+	}
+	return devicePath, nil
+}
+
+func (util *PortworxVolumeUtil) DetachVolume(u *portworxVolumeUnmounter, deviceName string) error {
+	hostName := u.plugin.host.GetHostName()
+	client, err := util.osdClient(hostName)
+	if err != nil {
+		return err
+	}
+
+	volumeID, err := getVolumeIDFromDeviceName(deviceName)
+	if err != nil {
+		return err
+	}
+
+	err = client.Detach(volumeID)
+	if err != nil {
+		glog.V(2).Infof("DetachVolume on %v failed with error %v", u.volumeID, err)
+		return err
+	}
+	return nil
+}
+
+func (util *PortworxVolumeUtil) MountVolume(m *portworxVolumeMounter, mountPath string) error {
+	hostName := m.plugin.host.GetHostName()
+	client, err := util.osdClient(hostName)
+	if err != nil {
+		return err
+	}
+
+	err = client.Mount(m.volumeID, mountPath)
+	if err != nil {
+		glog.V(2).Infof("MountVolume on %v failed with error %v", m.volumeID, err)
+		return err
+	}
+	return nil
+}
+
+func (util *PortworxVolumeUtil) UnmountVolume(u *portworxVolumeUnmounter, deviceName, mountPath string) error {
+	hostName := u.plugin.host.GetHostName()
+	client, err := util.osdClient(hostName)
+	if err != nil {
+		return err
+	}
+
+	volumeID, err := getVolumeIDFromDeviceName(deviceName)
+	if err != nil {
+		return err
+	}
+
+	err = client.Unmount(volumeID, mountPath)
+	if err != nil {
+		glog.V(2).Infof("UnmountVolume on mountPath: %v, VolName: %v failed with error %v", mountPath, u.volName, err)
+		return err
+	}
+	return nil
+}
+
+func getVolumeIDFromDeviceName(deviceName string) (string, error) {
+	if !strings.HasPrefix(deviceName, pxdDevicePrefix) {
+		return "", fmt.Errorf("Invalid DeviceName for Portworx Volume: %v", deviceName)
+	}
+	return strings.TrimPrefix(deviceName, pxdDevicePrefix), nil
+}
+
+func checkSnapInterval(snapInterval int) error {
+	if snapInterval == math.MaxUint32 || snapInterval < 0 ||
+		(snapInterval > 0 && snapInterval < 60) {
+		return fmt.Errorf(
+			"Interval must be greater than or equal to 60 and less than %v"+
+				" or 0 to disable scheduled snapshots",
+			math.MaxUint32)
+	}
+	return nil
+}
+
+func cosLevel(cos string) (osdapi.CosType, error) {
+	switch cos {
+	case "high", "3":
+		return osdapi.CosType_HIGH, nil
+	case "medium", "2":
+		return osdapi.CosType_MEDIUM, nil
+	case "low", "1":
+		return osdapi.CosType_LOW, nil
+	}
+	return osdapi.CosType_LOW,
+		fmt.Errorf("Cos must be one of %q | %q | %q", "high", "medium", "low")
+}
```

Comment:
Why is this required here when it is done in the backend? Maybe not convert params to a spec by pass it in the way docker does?